lib/trino-filesystem/src/main/java/io/trino/filesystem/cache/TrinoFileSystemCache.java

```diff
@@ -34,6 +34,12 @@ TrinoInput cacheInput(TrinoInputFile delegate, String key)
     TrinoInputStream cacheStream(TrinoInputFile delegate, String key)
             throws IOException;
 
+    /**
+     * Get the length of the TrinoInputFile, potentially using or updating the data cached at key.
+     */
+    long cacheLength(TrinoInputFile delegate, String key)
+            throws IOException;
+
     /**
      * Give a hint to the cache that the cache entry for location should be expired.
      */
```

Comment:
Maybe the javadoc should be changed noting that `location` could be a directory now. What's the semantics then?

Reply:
There is no "semantics", it's a hint to the cache, it can do whatever with it.

Reply:
I think the expectation is that it should evict cached data for all files in location, right?